src/simulation.rs

```diff
@@ -141,6 +142,45 @@ impl Simulation {
         // wait for report response
         let final_report = events_report.await.expect("events collection to end");
 
+        let ready_users = self.entities.values().filter_map(|entity| {
+            if let Entity::Ready { user } = entity {
+                Some(user)
+            } else {
+                None
+            }
+        });
+
+        let channel_stats = ready_users.clone().fold(
+            (0, usize::MAX, 0, HashSet::new()),
+            |(max, min, total_chans_of_users, mut channels_created), user| {
+                let u = user.try_read();
+
+                if let Ok(u) = u {
+                    log::debug!("getting user {} channel stats", u.localpart);
+                    let results = u.get_user_channels_stats((
+                        max,
+                        min,
+                        total_chans_of_users,
+                        &mut channels_created,
+                    ));
+                    log::debug!("channel stats - user {} - {:?}", u.localpart, results);
+                    (results.0, results.1, results.2, results.3.to_owned())
+                } else {
+                    (max, min, total_chans_of_users, channels_created)
+                }
+            },
+        );
+        let count = ready_users.count();
+        println!("CHANNEL STATS - ready users count: {}", count);
+        println!(
+            "CHANNELS STATS - \n max number user joined: {} \n  min number user joined : {} \n avg joined channel: {} \n total joined: {} \n total created: {}",
+            channel_stats.0,
+            channel_stats.1,
+            (channel_stats.2) as f64 / (count) as f64,
+            channel_stats.2,
+            channel_stats.3.len()
+        );
```

Comment:
Avoid printing everything inline in favor of adding this data to the `final_report`.